### PR TITLE
Convert Encoder callbacks to be boolean functions

### PR DESCRIPTION
**Pull request by drashna**
## Description

This converts the callback functions for encoders to be boolean, so that you can easily have code at the keyboard level that can overridden, or even that cannot (useful for oled/haptic/clicky behavior). 

## Types of Changes

- [x] Core
- [x] Bugfix
- [x] Enhancement/optimization

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cannonkeys/ortho75/ortho75.c [hunk 1]
```diff
@@ -44,6 +45,7 @@ void encoder_update_kb(uint8_t index, bool clockwise) {
     uint16_t held_keycode_timer = timer_read();
     register_code(mapped_code);
     while (timer_elapsed(held_keycode_timer) < MEDIA_KEY_DELAY){ /* no-op */ }
-    unregister_code(mapped_code);
+    unregistenr_code(mapped_code);
```
Comment (noroadsleft) [inline keyboards/cannonkeys/ortho75/ortho75.c:48]:
```suggestion
    unregister_code(mapped_code);
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Everything else in here is slight `__attribute__ ((weak))` :heavy_check_mark: for me.

**Review by noroadsleft — APPROVED** [review]
`__attribute__ ((weak))` :heavy_check_mark:

**Review by ridingqwerty — APPROVED** [review]
Everything here looks good to me.

**Comment by drashna** [comment]
(don't merge yet, rebasing and updating)

**Pull request merged**